streams/src/main/java/org/apache/kafka/streams/query/QueryConfig.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.query;
+
+public class QueryConfig {
+    final boolean collectExecutionInfo;
+    final boolean skipCache;
+
+    public QueryConfig(final boolean collectExecutionInfo, final boolean skipCache) {
```

Comment:
Instead of adding an extra boolean argument to the `query` method signature, I opted for this since this is future-forward in case we want to add more configurations